code/Util/DamageInfoExtensions.cs

```diff
@@ -8,15 +8,36 @@ public static class DamageInfoExtensions
 	/// Is this damage a headshot?
 	/// </summary>
 	public static bool IsHeadshot( this DamageInfo info ) => info.Hitbox.HasTag( "head" );
+
 	/// <summary>
 	/// If this damage ends up being lethal, should the player scream when killed?
 	/// </summary>
 	public static bool IsSilent( this DamageInfo info ) => info.HasTag( DamageTags.Silent );
+
 	/// <summary>
 	/// If this damage was avoidable e.g. a traitor dying to a
 	/// teammate's C4, then no karma penalty will be given to the teammate.
 	/// </summary>
 	public static bool IsAvoidable( this DamageInfo info ) => info.HasTag( DamageTags.Avoidable );
+
+	/// <summary>
+	/// Creates a bloodsplatter effect in the world.
+	/// </summary>
+	/// <param name="info"></param>
+	/// <param name="player">The attacked player, their blood is getting splattered.</param>
+	/// <param name="maxDistance">How far we will check for a place to splatter.</param>
+	public static void CreateBloodSplatter( this DamageInfo info, Player player, float maxDistance )
+	{
+		var splatterTrace = Trace.Ray( new Ray( info.Position, info.Force.Normal ), maxDistance )
+			.Ignore( player )
```

Comment:
Could maybe use `WorldOnly()` or whatever it is called, I'm not sure how the decals behave when placed on a player or entity.

Reply:
From my testing they don't work on Players. It should probably include Entities for props in the world. I guess it's fine if blood only appears on static WorldEntity geometry? But there might be situations where it won't appear because it was going to hit a big refrigerator or something.

Reply:
let me mess around with it and ill let u know

Reply:
yeah should be good mb